app/models/generic_object_definition.rb

```diff
@@ -7,6 +16,18 @@ class GenericObjectDefinition < ApplicationRecord
   has_many  :generic_objects
 
   def defined_attributes
-    @defined_attributes ||= properties[:attributes].stringify_keys
+    properties[:attributes]
+  end
+
+  def properties=(props)
+    props = props.symbolize_keys
+    if props.key?(:attributes)
+      # unify attribute name as string and attribute type as symbol
+      props[:attributes] = props[:attributes].each_with_object({}) do |(name, type), memo|
+        raise ArgumentError, "#{type} is not recognized type" unless TYPE_MAP.key?(type.to_sym)
+        memo[name.to_s] = type.to_sym
+      end
```

Comment:
Also note that the caller should throw props away since it's no longer valid after this (that is, the model will have a completely different hash than the caller passed to `properties=`), and if it's modified, it won't be the same.